src/compat/eme/generate_key_request.ts

```diff
@@ -79,8 +94,35 @@ export function patchInitData(initData : Uint8Array) : Uint8Array {
         initData[offset + 26] === 0xFB &&
         initData[offset + 27] === 0x4B
     ) {
-      log.info("Compat: CENC PSSH found.");
-      cencs = concat(cencs, currentPSSH);
+      const cencOffsets = getNextBoxOffsets(currentPSSH);
+      const version = cencOffsets === null ? undefined :
+                                             currentPSSH[cencOffsets[1]];
+      log.info("Compat: CENC PSSH found with version", version);
+      if (version === undefined) {
```

Comment:
Ever considered using an if-else-if construct ? I know some people find it less readable. It is probably a matter of taste (and C-coding habits) ...
```suggestion
if (version === undefined) {
    log.warn("Compat: could not read version of CENC PSSH");
} else if (encounteredCencs.length === 0) {
    encounteredCencs.push({ version, pssh: currentPSSH });
} else if (hasV1 = encounteredCencs.some(cenc => cenc.version === 1),
           isV1 = version === 1,
           hasV1 == isV1 ) {
    encounteredCencs.push({ version, pssh: currentPSSH });
} else if ( isV1 ) {
    log.warn("Compat: cenc version 1 encountered, removing every other cenc pssh box.");
    encounteredCencs = [{ version, pssh: currentPSSH }];
} else {
    log.warn("Compat: filtering out cenc pssh box with wrong version", version);
}
```